### PR TITLE
[Triton-MLIR][ROCM] First steps to add ROCm support.

### PR DESCRIPTION
**Pull request by dfukalov**
Merge from upstream and

first steps to add ROCm support to triton-mlir branch:
1. Added USE_ROCM and DEBUG_ROCM definitions to add ROCDL dialect support and switch off some NVPTX related output generation
2. Added ROCDL passes to pipeline
3. Temporarily hacked tests since switch for ROCm mode is not introduced yet.

### CODE REVIEW

**Comment by dfukalov** [comment]
If the upstream merge should be separated, I can re-create two other PRs

**Comment by whchung** [comment]
@dfukalov Please help paste unit test results.

File: CMakeLists.txt [hunk 1]
```diff
@@ -132,18 +146,23 @@ endif()
 # Python module
 if(TRITON_BUILD_PYTHON_MODULE)
     message(STATUS "Adding Python module")
+    set(PYTHON_SRC_PATH ${CMAKE_CURRENT_SOURCE_DIR}/python/src)
```
Comment (whchung) [inline CMakeLists.txt:149]: Some of these changes don't seem to be ROCm-specific. Could you help explain the rationale behind the changes?

**Review by whchung — CHANGES_REQUESTED** [review]
Please help separate upstream merge into a separate PR so it can be easier to understand where ROCm-specific changes are applied. And please help provide *FOOLPROOF* step-by-step instructions so I can follow and reproduce the current results.

File: test/Conversion/tritongpu_to_llvm.mlir [hunk 2]
```diff
@@ -3,7 +3,7 @@
 module attributes {"triton_gpu.num-warps" = 4 : i32} {
   // CHECK: llvm.func @test_empty_kernel(%arg0: i32, %arg1: !llvm.ptr<f16, 1>)
   // Here the 128 comes from the 4 in module attribute multiples 32
-  // CHECK:  attributes {nvvm.kernel = 1 : ui1, nvvm.maxntid = 128 : i32} {{.*}}
+  // XHECK:  attributes {nvvm.kernel = 1 : ui1, nvvm.maxntid = 128 : i32} {{.*}}
```
Comment (whchung) [inline test/Conversion/tritongpu_to_llvm.mlir:6]: typo

**Comment by whchung** [comment]
Adding @rsanthanam-amd to help understand the bar I am setting.

File: CMakeLists.txt (hunk 1 above)
Comment (whchung) [reply CMakeLists.txt:149]: From the look of it, it's from upstream merge. In order to reduce confusion, please separate upstream merge into another PR. Keep ROCm-specific changes here.

**Pull request closed without merging**